scripts/030_extract_sipnet_output.R

```diff
@@ -111,152 +123,49 @@ ens_results <- furrr::future_pmap_dfr(
         ) |>
             dplyr::mutate(
                 site_id = .env$site_id,
-                parameter = as.numeric(.env$ens)
+                parameter = as.numeric(.env$ens),
+                pft = .env$pft,
+                base_site_id = .env$base_site_id
             ) |>
             dplyr::rename(time = posix)
     },
     # Avoids warning "future unexpectedly generated random numbers",
-    # Fixed in units v 0.9.0
     # .Bug report: https://github.com/r-quantities/units/issues/409
+    # Fixed in units version >= 0.8.7
     .options = furrr::furrr_options(seed = TRUE)
-) |>
-    dplyr::group_by(parameter, site_id, year) |>
-    # filter(year <= end_year) |> # not sure why this was necessary;
-    # should be taken care of by read.output
-    dplyr::filter(time == max(time)) |> # only take last value
+) 
+
+ens_results <- ens_results_raw |>
+    dplyr::group_by(parameter, base_site_id, pft, year) |>
     dplyr::ungroup() |>
-    dplyr::arrange(parameter, site_id, year) |>
+    dplyr::arrange(parameter, base_site_id, pft, year) |>
     tidyr::pivot_longer(
-        cols = all_of(variables),
+        cols = tidyr::all_of(variables),
         names_to = "variable",
         values_to = "prediction"
     ) |>
     dplyr::rename(datetime = time) |>
-    dplyr::left_join(design_points, by = "site_id") |> 
-    dplyr::select(datetime, site_id, lat, lon, pft, parameter, variable, prediction)
-
+    dplyr::left_join(site_meta,
+        by = c("site_id", "base_site_id", "pft")
+    ) |>
+    dplyr::select(-site_id) |>
+    dplyr::rename(site_id = base_site_id) |>
+    dplyr::select(datetime, site_id, lat, lon, pft, parameter, variable, prediction) |>
+    dplyr::mutate(datetime = lubridate::floor_date(datetime, unit = "month")) |>
+    dplyr::group_by(datetime, site_id, lat, lon, pft, parameter, variable) |>
+    dplyr::summarise(
+        prediction = mean(prediction, na.rm = TRUE), .groups = "drop"
```

Comment:
Might be worth a note that this is valid for pool sizes but not (or at least hard to interpret) for flux variables